assets/js/modules/analytics/datastore/setup-flow.test.js

```diff
@@ -0,0 +1,327 @@
+/**
+ * `modules/analytics` data store: setup-flow tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { createRegistry } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { MODULES_ANALYTICS } from './constants';
+import { createTestRegistry, unsubscribeFromAll } from 'tests/js/utils';
+import * as modulesAnalytics from '../';
+import { MODULES_ANALYTICS_4 } from '../../analytics-4/datastore/constants';
+
+const accountID = 'foo-bar';
+
+const populateAnalyticsDatastore = ( registry ) => {
+	registry.dispatch( MODULES_ANALYTICS ).receiveGetProperties(
+		[
+			{
+				// eslint-disable-next-line sitekit/acronym-case
+				accountId: '151753095',
+				id: 'UA-151753095-1',
+				name: 'rwh',
+			},
+			{
+				// eslint-disable-next-line sitekit/acronym-case
+				accountId: '151753095',
+				id: 'UA-151753095-1',
+				name: 'troubled-tipped.example.com',
+			},
+
+		],
+		{ accountID }
+	);
+};
+
+const populateAnalytics4Datastore = ( registry ) => {
+	registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetProperties( [
+		{
+			_id: '1000',
+			_accountID: '100',
+			name: 'properties/1000',
+			createTime: '2014-10-02T15:01:23Z',
+			updateTime: '2014-10-02T15:01:23Z',
+			parent: 'accounts/100',
+			displayName: 'Test GA4 Property',
+			industryCategory: 'TECHNOLOGY',
+			timeZone: 'America/Los_Angeles',
+			currencyCode: 'USD',
+			deleted: false,
+		},
+	],
+	{ accountID },
+	);
+	registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetWebDataStreams(
+		[
+			{
+				_id: '2000',
+				_propertyID: '1000',
+				name: 'properties/1000/webDataStreams/2000',
+				// eslint-disable-next-line sitekit/acronym-case
+				measurementId: '1A2BCD345E',
+				// eslint-disable-next-line sitekit/acronym-case
+				firebaseAppId: '',
+				createTime: '2014-10-02T15:01:23Z',
+				updateTime: '2014-10-02T15:01:23Z',
+				defaultUri: 'http://example.com',
+				displayName: 'Test GA4 WebDataStream',
+			},
+		],
+		{ propertyID: 'foobar' }
```

Comment:
@tofumatt   Looks like I've already started ruining the code base with some PRs of mine merged with meaningless properties :see_no_evil: 

Happy to fix them in this PR if you think it's worth it! (or not because scope creep and this PR is already large enough to lose context!)

Reply:
I fixed this instance though with a propertyID taken from other GA4 tests